backend/controllers/amendmentCaseNote.ts

```diff
@@ -43,12 +51,10 @@ export default ({ prisonApi, caseNotesApi, logError }) => {
     }
   }
 
-  const stashSateAndRedirectToIndex = (req, res, moreDetail, errors) => {
-    req.flash('amendmentErrors', errors)
-    req.flash('formValues', {
-      moreDetail,
-    })
-    return res.redirect(req.originalUrl)
+  const stashStateAndRedirectToAmendCaseNote = (req, res, moreDetail, errors, offenderNo, caseNoteId) => {
+    if (errors.length > 0) req.flash('amendmentErrors', errors)
+    req.flash('formValues', { moreDetail })
+    return res.redirect(`/prisoner/${offenderNo}/case-notes/amend-case-note/${caseNoteId}`)
```

Comment:
since there is a confirm page need to specify where to go back to in case of problems